python/xorbits/_mars/dataframe/sort/psrs.py

```diff
@@ -660,15 +660,77 @@ def _execute_series_map(cls, ctx, op):
                 values = a.iloc[poses[i] : poses[i + 1]]
                 ctx[out.key, (i,)] = values
 
+    @classmethod
+    def _calc_pivots_poses(
+        cls, index, pivots, is_multi_index=False, level=None, sort_remaining=True
+    ):
+        if not is_multi_index:
+            return index.searchsorted(list(pivots), side="right")
+
+        if index.nlevels != pivots.nlevels:
+            raise TypeError(
```

Comment:
A `ValueError` may be more suitable. See https://docs.python.org/3/library/exceptions.html#ValueError for the meaning of builtin Errors.

Reply:
done, thx